Add Ladle stories, and performed some refactoring

Yukinobu-Mine opened the pull request:
*Issue #, if available:*

*Description of changes:*
Refactor `ChatPage`, `ChatListDrawer` and related components.

useChat
- Add `conversationError`
  - Move the error handling code to `ChatPage`
- Remove dependencies
  - `useNavigate`
  - `useTranslation`
  - `useSnackbar`

ChatMessage
- Move the action handling code to `ChatPage`
  - Submit feedback
- Move the code to get related documents to `ChatPage`
- Remove dependencies
  - `useChat`

InputChatContent
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - hasContinue / disabledContinue
  - isLoading
- Remove dependencies
  - `useChat`

TextInputChatContent
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - isLoading
- Remove dependencies
  - `useChat`

ChatListDrawer
- Move the action handling code to `AppContent`
  - Delete conversation
  - Update conversation title
- Move the appearance control code to `AppContent`
  - isAdmin
- Remove dependencies
  - `useNavigate`
  - `useUser`

Menu
- Move the action handling code to `AppContent`
  - Select language
  - Clear conversations
- Remove dependencies
  - `useNavigate`
  - `useConversation`

AppContent
- Accept the action handling code from `Menu` and `ChatListDrawer`
  - Select language
  - Clear conversations
  - Delete conversation
  - Update conversation title

ChatPage
- Accept the code to get related documents from `ChatMessage`
- Accept the appearance control code from `InputChatContent` and `TextInputChatContent`
- Accept the error handling code from `useChat`

Add Ladle stories
- Agent
  - InputChatContent
  - InputChatContentHasRegenerate
  - ProcessingIndicator
  - Tools
- ChatListDrawer (using msw: mock service wrapper)
  - Admin
  - NonAdmin
- ChatMessage
  - Conversation
- Helper
  - Shortcuts
- InputChatContent
  - Ideal
  - HasRegenerate
  - HasContinue
- InputText
  - Text
  - URL
  - Date
  - Number
- ListItemBot
  - MyBots
  - RecentlyUsedSharedBots
  - ApiManagement
  - Analytics
- ModalDialog
  - AddApiKey
  - ClearConversations
  - DeleteApi
  - DeleteApiKey
  - DeleteBot
  - DeleteConversation
  - Feedback
  - InstructionsSamples
  - SelectLanguage
    - Performed refactoring
  - ShareBot
- PopoverMenu
  - ChatHeader
- Progress
- Select
  - Ideal
  - Clearable
- Textarea
  - Ideal
  - NoBorder

Fixed story name
- Skelton -> Skeleton
  - Indeal -> Ideal

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Yukinobu-Mine committed:
Add Ladle stories
Added stories
- Helper
  - Shortcuts
- InputText
  - Text
  - URL
  - Date
  - Number
- ListItemBot
  - MyBots
  - RecentlyUsedSharedBots
  - ApiManagement
  - Analytics
- ModalDialog
  - AddApiKey
  - ClearConversations
  - DeleteApi
  - DeleteApiKey
  - DeleteBot
  - DeleteConversation
  - Feedback
  - InstructionsSamples
  - SelectLanguage
    - Performed refactoring
  - ShareBot
- PopoverMenu
  - ChatHeader
- Progress
- Select
  - Ideal
  - Clearable
- Textarea
  - Ideal
  - NoBorder

Fixed story name
- Skelton -> Skeleton
  - Indeal -> Ideal

Yukinobu-Mine committed:
Refactor `ChatPage`, `ChatListDrawer` and related components
`useChat`
- Add `conversationError`
  - Move the error handling code to `ChatPage`
- Remove dependencies
  - `useNavigate`
  - `useTranslation`
  - `useSnackbar`

`ChatMessage`
- Move the action handling code to `ChatPage`
  - Submit feedback
- Move the code to get related documents to `ChatPage`
- Remove dependencies
  - `useChat`

`InputChatContent`
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - hasContinue / disabledContinue
  - isLoading
- Remove dependencies
  - `useChat`

`TextInputChatContent`
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - isLoading
- Remove dependencies
  - `useChat`

`ChatListDrawer`
- Move the action handling code to `AppContent`
  - Delete conversation
  - Update conversation title
- Move the appearance control code to `AppContent`
  - isAdmin
- Remove dependencies
  - `useNavigate`
  - `useUser`

`Menu`
- Move the action handling code to `AppContent`
  - Select language
  - Clear conversations
- Remove dependencies
  - `useNavigate`
  - `useConversation`

`AppContent`
- Accept the action handling code from `Menu` and `ChatListDrawer`
  - Select language
  - Clear conversations
  - Delete conversation
  - Update conversation title

`ChatPage`
- Accept the code to get related documents from `ChatMessage`
- Accept the appearance control code from `InputChatContent` and `TextInputChatContent`
- Accept the error handling code from `useChat`

Add Ladle stories
- Agent
  - InputChatContent
  - InputChatContentHasRegenerate
  - ProcessingIndicator
  - Tools
- ChatListDrawer (using msw: mock service wrapper)
  - Admin
  - NonAdmin
- ChatMessage
  - Conversation
- InputChatContent
  - Ideal
  - HasRegenerate
  - HasContinue

Yukinobu-Mine committed:
Fix eslint warnings
ChatMessage.tsx
- react-hooks/exhaustive-deps

ListItemBot.stories.tsx
- tailwindcss/enforces-shorthand

Yukinobu-Mine committed:
Fixed eslint errors
ChatListDrawer
- Unused eslint-disable directive

ChatMessage
- Unused eslint-disable directive

wadabee commented on frontend/src/features/agent/components/TextInputChatContent.tsx:
```diff
@@ -17,25 +16,23 @@ import { BaseProps } from '../../../@types/common';
+  disabledRegenerate?: boolean;
   disabled?: boolean;
   placeholder?: string;
+  hasRegenerate: boolean;
```
has より can の方がわかりやすそうかなと思ったんですが、いかがでしょうか？

wadabee commented on frontend/src/components/InputChatContent.tsx:
```diff
@@ -37,9 +36,14 @@ import {
   disabled?: boolean;
   placeholder?: string;
   dndMode?: boolean;
+  hasRegenerate: boolean;
+  hasContinue: boolean;
```
別のところにも書いたのですが、has より can の方がわかりやすそうかなと思ったんですが、いかがでしょうか？

wadabee commented on frontend/src/pages/ChatPage.tsx:
```diff
@@ -395,8 +415,29 @@ const ChatPage: React.FC = () => {
                         <ChatMessage
                           chatContent={message}
+                          relatedDocuments={
+                            (() => {
+                              if (message.usedChunks) {
+                                // usedChunks is available for existing messages
+                                return message.usedChunks.map(chunk => ({
+                                  chunkBody: chunk.content,
+                                  contentType: chunk.contentType,
+                                  sourceLink: chunk.source,
+                                  rank: chunk.rank,
+                                }));
+                              } else {
+                                // For new messages, get related documents from the api
+                                return getRelatedDocuments(message.id);
+                              }
+                            })()
```
支障がなければuseMemoでメモ化をお願いします！

Yukinobu-Mine replied:
@wadabee 一応 `ChatMessage` 側で受け取った `relatedDocuments` を[メモ化している](https://github.com/aws-samples/bedrock-claude-chat/pull/485/files#diff-d9bc5cfc88dfde0d963ccfc3c4405bc742f3a029a411195706eca36aee8cb554R40)のですが、このやり方はちょっと違った感じでしょうか?
`ChatPage` 側で、`messages` 内の各メッセージの `relatedDocuments` をメモ化する必要がある場合、どのような形でメモ化するのが望ましいですか?

wadabee reviewed (changes requested):
軽微なコメントを記載したので、ご確認をお願いしますmm

Yukinobu-Mine committed:
Rename some properties of `InputChatContent`, `TextInputChatContent`
- hasRegenerate -> canRegenerate
- hasContinue -> canContinue

Yukinobu-Mine commented:
@wadabee has の can へのリネームを行いました。useMemo に関して質問させて頂いたので、ご確認お願いします 🙇

Yukinobu-Mine committed:
Memorize related documents of messages in `ChatPage`.
- Add memorized `ChatMessage` as `ChatMessageWithRelatedDocuments`.

Yukinobu-Mine committed:
Refactor `ChatListDrawer` and related components.
`ChatListDrawer`
- Move the code to get bots and conversations to `AppContent`
  - conversations
  - starredBots
  - recentlyUsedUnsterredBots
- Remove dependencies
  - `useConversation`
  - `useBot`

`AppContent`
- Accept the code to get bots and conversations from `ChatListDrawer`
  - conversations
  - starredBots
  - recentlyUsedUnsterredBots

wadabee reviewed (approved):
LGTM!!
